datastore/src/main/java/com/ds/datastore/BookController.java

```diff
@@ -58,21 +59,42 @@ public BookController(BookRepository repository, BookModelAssembler assembler, B
     @PostMapping("/bookstores/book")
     protected CollectionModel<EntityModel<Book>> oneBookToManyStores(@RequestBody Book book, @RequestParam List<String> id) throws Exception {
         List<EntityModel<Book>> entityList = new ArrayList<>();
-        for(String storeId: id)
-        {
-            book.setStoreID(Long.parseLong(storeId));
-            if(!this.map.containsKey(Long.parseLong(storeId))) continue;
-            HttpURLConnection con = createConnection(this.map.get(Long.parseLong(storeId)) + "/books", "POST");
+        if (!this.storeRepository.findAll().get(0).getServerId().equals(this.leader.getLeader())) {
+            String address = this.map.get(this.leader.getLeader());
+            address = address.substring(0,address.lastIndexOf("/") + 1) + "book?id=" + id.toString().replaceAll("[\\[ \\]]", "");
```

Comment:
Lets use `String.join(", ", list);` ... this is painful :)

Reply:
Who doesn't like a good regex?

Reply:
LOL

https://www.onely.com/wp-content/uploads/blog/2017/principles-regular-expressions-seo/003-picard-now-i-have-two-problems.jpg